viadot/sources/base.py

```diff
@@ -124,6 +124,47 @@ def to_excel(
         out_df.to_excel(path, index=False, encoding="utf8")
         return True
 
+    def to_parquet(
+        self,
+        path: str,
+        if_exists: Literal["append", "replace", "skip"] = "replace",
+        if_empty: str = "warn",
+        **kwargs,
+    ) -> None:
+        """
+        Write from source to a Parquet file.
+
+        Args:
+            path (str): The destination path.
+            if_exists (str, optional): What to do if the file exists. Defaults to "replace".
+            if_empty (str, optional): What to do if the source contains no data. Defaults to "warn".
+
+        """
+        try:
+            df = self.to_df(if_empty=if_empty)
+        except SKIP:
+            return False
+        if if_exists == "append" and os.path.isfile(path):
+            parquet_df = pd.read_parquet(path)
+            out_df = pd.concat([parquet_df, df])
+        elif if_exists == "replace":
+            out_df = df
+        elif if_exists == "skip":
+            logger.info("Skipped.")
+            return
+        else:
+            out_df = df
+
+        # create directories if they don't exist
+        try:
+            if not os.path.isfile(path):
+                directory = os.path.dirname(path)
+                os.makedirs(directory, exist_ok=True)
+        except:
```

Comment:
Please add specific exceptions. Why there is a `pass`?

Reply:
There is more options. I updated docstring to make it clear.